### PR TITLE
[RFC] EB-3601 Fix QUERY_STRING in url

### PR DESCRIPTION
**Pull request by zaizai22222**

### CODE REVIEW

File: lib/ezutils/classes/ezhttptool.php [hunk 1]
```diff
@@ -585,9 +585,17 @@ static function createRedirectUrl( $path, $parameters = array() )
         if ( $port )
             $uri .= ':' . $port;
         $uri .= $path;
-        
-        if ( isset($_SERVER[QUERY_STRING]) && $_SERVER[QUERY_STRING] != '' )
-        	$uri .= '?' . $_SERVER[QUERY_STRING];
+
+        if ( isset($_SERVER['QUERY_STRING']) && $_SERVER['QUERY_STRING'] != '' ) {
+            //If the QUERY_STRING only has "redirect" parameter, do not show it in url
+            if (strpos($_SERVER['QUERY_STRING'],'redirect') === false || strpos($_SERVER['QUERY_STRING'],'&')!== false ) {
```
Comment (baruica) [inline lib/ezutils/classes/ezhttptool.php]: I would have used parse_url() to make sure "redirect" is the only query parameter in the url, looking for "&" is a lot less readable here...

**Comment by zaizai22222** [comment]
@baruica @trompette Please review

File: lib/ezutils/classes/ezhttptool.php [hunk 2]
```diff
@@ -585,9 +585,20 @@ static function createRedirectUrl( $path, $parameters = array() )
...
+            if ($parsedQuery !== null) {
+                $uri = str_replace( '?'. $parsedQuery, '', $uri);
+            }
+
+            parse_str($parsedQuery, $uriQueryArray);
+            parse_str($_SERVER['QUERY_STRING'], $serverQueryArray);
+            $queryArray = array_merge($serverQueryArray, $uriQueryArray);
+            unset($queryArray['redirect']);
+
+            $uri = $uri . '?' . http_build_query($queryArray);
```
Comment (baruica) [inline lib/ezutils/classes/ezhttptool.php:600]: .= operator could have been used here

**Comment by baruica** [comment]
CR OK

**Comment by trompette** [comment]
CR OK.

**Pull request merged**